### PR TITLE
[x86] Clear the fp stack when exceptions are thrown. Fixes #45841.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/tests/Makefile.am [hunk 1]
```diff
@@ -789,7 +789,8 @@ TEST_IL_SRC_UNIVERSAL=			\
 	delegate-with-null-target.il	\
 	bug-318677.il	\
 	gsharing-valuetype-layout.il	\
-	invalid_generic_instantiation.il
+	invalid_generic_instantiation.il \
+	bug-45841-fpstack-exceptions.il
```
Comment (marek-safar) [inline mono/tests/Makefile.am]: You are missing the file

File: mono/mini/exceptions-x86.c [hunk 2]
```diff
@@ -536,7 +536,7 @@ mono_x86_resume_unwind (mgreg_t *regs, MonoObject *exc,
 static guint8*
 get_throw_trampoline (const char *name, gboolean rethrow, gboolean llvm, gboolean corlib, gboolean llvm_abs, gboolean resume_unwind, MonoTrampInfo **info, gboolean aot)
 {
-	guint8 *start, *code;
+	guint8 *start, *code, *labels [16];
```
Comment (lewurm) [inline mono/mini/exceptions-x86.c]: `s/16/2` ?

Comment (vargaz) [inline mono/mini/exceptions-x86.c]: less chance to forget to increase it if the code is changed.

**Comment by akoeplinger** [comment]
@vargaz the new test fails consistently on the 64bit Wrench lane, but interestingly works fine on the 64bit Jenkins OSX bots.

**Comment by vargaz** [comment]
The test started to fail after:
https://wrench.internalx.com/Wrench/ViewLane.aspx?lane_id=2457&host_id=148&revision_id=820023

**Pull request merged**